chainerx/_cuda.py

```diff
@@ -0,0 +1,38 @@
+import chainerx
+
+try:
+    # _pybind_cuda is unavailable if ChainerX is built without CUDA.
+    from chainerx import _pybind_cuda
+    _available = True
+except Exception:
+    _available = False
+
+try:
+    import cupy
+    _cupy_available = True
+except Exception:
+    _cupy_available = False
+
+_chainerx_allocator = None
+
+
+def cupy_share_allocator(owner=chainerx._global_context):
```

Comment:
Is there any reason to make `owner` customizable?

Reply:
I don't have a strong opinion since contexts aren't tested much but I see a couple of reasons. Do you have any opinions to why it should not be?
- As this feature is "experimental" I think it's okay to allow some degree of freedom. Of course, the degree can be discussed, but my understanding is that it is difficult at this point. 
- It default to the default global context and does not make it more difficult to use.
- To allow users to create new contexts and still use this feature.

Reply:
I believe the CUDA backend instance is *always* used as the owner and there will never be a requirement to change that.

Reply:
Yes. What's the concern?
Even if it's the context instead of the backend, my argument above does not change (no need to customize it).

Reply:
Okay I removed it as you suggested.